src/js/models/opportunities.js

```diff
@@ -72,15 +71,26 @@ export default class Opportunities{
 
 	async updateJuniorStatus(oppId,juniorId,decision){
 		try{
-			let docId = '';
+			let oppDocId = '';
+			let profileDocId = '';
+			//update firebase NewOpportunities register status
 			await db.collection('NewOpportunities').doc(oppId).collection("registered").where("uid","==",juniorId).get().then(docs => {
 				docs.forEach(doc => {
-					docId = doc.id;
+					oppDocId = doc.id;
 				})				
 			})
-			await db.collection('NewOpportunities').doc(oppId).collection("registered").doc(docId).update({
+			await db.collection('NewOpportunities').doc(oppId).collection("registered").doc(oppDocId).update({
 				status:decision
-			})			
+			})

```

Comment:
Can this be simplified to one call like https://stackoverflow.com/questions/53836195/firebase-functions-update-all-documents-inside-a-collection?

Reply:
No. oppDocId has to be recorded to be used in a different function. Can't use async function in forEach.

Reply:
> oppDocId has to be recorded to be used in a different function

Which other function is it used in?

> Can't use async function in forEach.

The solution in that link does not use async/await so you can see how to do it [here](https://youtu.be/Jr7pDZ1RAUg?t=227). This solution also does not use async function in forEach

Reply:
> Can't use async function in forEach

[What happened here then](https://github.com/Yuxuan-Lin/VioLegacy/blob/4ad6fcdfb60b7adca990c697f7f2eaf59a6c5fc5/src/js/models/messages.js#L19)

Reply:
Done